server/service/apple_mdm.go

```diff
@@ -3855,20 +3860,21 @@ func NewInstalledApplicationListResultsHandler(
 
 		}
 
-		switch poll {
-		case true:
+		if poll {
+			// Queue a job to verify the VPP install.
 			err := worker.QueueVPPInstallVerificationJob(ctx, ds, logger, worker.VerifyVPPTask, verifyRequestDelay, installedAppResult.HostUUID(), installedAppResult.UUID())
 			if err != nil {
 				return ctxerr.Wrap(ctx, err, "InstalledApplicationList handler: queueing vpp install verification job")
 			}
-		case false:
-			if err := ds.RemoveHostMDMCommand(ctx, fleet.HostMDMCommand{CommandType: fleet.VerifySoftwareInstallVPPPrefix, HostID: hostID}); err != nil {
-				return ctxerr.Wrap(ctx, err, "InstalledApplicationList handler: removing host mdm command")
-			}
-
 		}
 
-		return nil
+		// If we get here, we're in a terminal state, so we can remove the verify command.
+		return ctxerr.Wrap(
```

Comment:
thx @gillespi314 for the reminder that `ctxerr.Wrap` returns `nil` if the error is `nil`!